clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -162,6 +188,19 @@ public synchronized void update(Cluster cluster, long now) {
         this.lastSuccessfulRefreshMs = now;
         this.version += 1;
 
+        // Handle expiry of topics from the metadata list
+        Iterator<Map.Entry<String, Long>> it = topics.entrySet().iterator();
+        while (it.hasNext()) {
+            Map.Entry<String, Long> entry = it.next();
+            Long expireMs = entry.getValue();
+            if (expireMs == TOPIC_EXPIRY_NEEDS_UPDATE)
+                entry.setValue(now + metadataExpireMs);
```

Comment:
Have added a comment. Using a different hard-coded value from refresh interval would require a different mechanism for scheduling expiry when the hard-coded value is not a multiple of the refresh interval. Since the topics are actually used for refresh, using a single timer made sense.